modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -157,13 +159,90 @@ export default class GeoJsonLayer extends CompositeLayer {
     const LineStringsLayer = this.getSubLayerClass('line-strings', PathLayer);
     const PointsLayer = this.getSubLayerClass('points', ScatterplotLayer);
 
+    const layerProps = {
+      points: {data: undefined, _dataDiff: undefined},
+      lines: {data: undefined, _dataDiff: undefined},
+      polygons: {data: undefined, _dataDiff: undefined},
+      polygonsOutline: {data: undefined, _dataDiff: undefined}
+    };
+
+    const isBinary = this._isBinary();
+
+    if (isBinary) {
+      const {points, lines, polygons} = this.props.data;
+
+      layerProps.points.data = {
```

Comment:
This creates a new `data` object on every rerender and will recalculate all attributes if any prop changes. `data` should be created once inside `updateState` on `changeFlags.dataChange`.